### PR TITLE
SM and SMZ3 apworld support

### PR DESCRIPTION
**Pull request by lordlou**
## What is this fixing or adding?
Added support for apworld zip archive to SM and SMZ3.

## How was this tested?
I generated a multiworld with SM and SMZ3 on source and on a built Archipelago package.

### CODE REVIEW

File: worlds/sm/__init__.py [hunk 1]
```diff
@@ -279,14 +280,14 @@ def APPrePatchRom(self, romPatcher):
         # first apply the sm multiworld code patch named 'basepatch' (also has empty tables that we'll overwrite),
         #  + apply some patches from varia that we want to be always-on.
         # basepatch and variapatches are both generated from https://github.com/lordlou/SMBasepatch
-        romPatcher.applyIPSPatch(os.path.join(os.path.dirname(__file__),
-                                              "data", "SMBasepatch_prebuilt", "multiworld-basepatch.ips"))
-        romPatcher.applyIPSPatch(os.path.join(os.path.dirname(__file__),
-                                              "data", "SMBasepatch_prebuilt", "variapatches.ips"))
+        romPatcher.applyIPSPatch("/".join((str(appDir),
```
Comment (Berserker66) [inline worlds/sm/__init__.py:283]: AP injects a `__file__` attribute on your world, which might be what you want? pkgutil would be better

**Review by Berserker66 — APPROVED** [review]

**Comment by Berserker66** [comment]
For some reason my PyCharm refuses to Checkout this PR, but on visual inspection it looks ok.

File: worlds/sm/__init__.py (hunk 1 above)
Comment (lordlou) [reply worlds/sm/__init__.py]: I aim to minimize changes done to the base code to ease future update of VARIA. Using pkgutil.get_data would require more changes. As for World.__file__, I'll use it here but trying to import from __init__ in deeper packages results in circular dependencies. I would need to pass down the value, which I'm not fond of.

Maybe I'm missing something else?

**Comment by lordlou** [comment]
I'm using VS Code so cant really help you there. I did the PR the same way I've always done the others so thats strange.

**Pull request merged**